### PR TITLE
add pagination for internship page (fixes #164)

### LINKED ISSUE #164: [Feature Request]: Pagination for internship page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

There are many internship opportunities in the database. Scrolling down extensively negatively impacts the user experience. Therefore, I can create pagination for viewing internships.

### Use Case

By breaking down content into smaller, manageable parts, users can navigate through information more easily without feeling overwhelmed.

### Benefits

Users can quickly jump to specific sections of content, making it easier to find what they're looking for without excessive scrolling.

### Add ScreenShots

I like to design a pagination like below.

image

In the top i will display no of results showing out of total

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by DharshiBalasubramaniyam**
This pr fixes #164.

screenshots:
Screenshot 2024-05-17 093744

Screenshot 2024-05-17 093803

video link for my demonstration:
https://drive.google.com/file/d/18jspsXz9p51kh5shfldUN8_YlWuyF4G3/view?usp=sharing

### CODE REVIEW

File: src/Componet/Home.jsx [hunk 1]
```diff
@@ -160,7 +160,7 @@ function Home() {
           </button>
         </div>
       </div> */}
```
Comment (Shubh942) [inline src/Componet/Home.jsx:162]: Remove the comments of the code.

**Comment by Shubh942** [comment]
Looks good. @HimanshuNarware check it out.

**Comment by PSS2134** [comment]
Looks Good @DharshiBalasubramaniyam , Good Work !

**Comment by DharshiBalasubramaniyam** [comment]
> Looks Good @DharshiBalasubramaniyam , Good Work !

Thank you

**Comment by DharshiBalasubramaniyam** [comment]
@PSS2134 Can you please add gssoc and level label to this pr.

**Pull request merged**